pkg/backup/compaction_test.go

```diff
@@ -675,6 +677,89 @@ func TestBackupCompactionUnsupportedOptions(t *testing.T) {
 	}
 }
 
+func TestCompactionCheckpointing(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	// Need a large enough backup so that it doesn't finish in one iteration
+	const numAccounts = 10000
+	var mu syncutil.Mutex
+	manifestNumFiles := -1
+	tc, db, _, cleanup := backupRestoreTestSetupWithParams(
+		t, singleNode, numAccounts, InitManualReplication, base.TestClusterArgs{
+			ServerArgs: base.TestServerArgs{
+				Knobs: base.TestingKnobs{
+					BackupRestore: &sql.BackupRestoreTestingKnobs{
+						AfterLoadingCompactionManifestOnResume: func(m *backuppb.BackupManifest) {
+							mu.Lock()
+							defer mu.Unlock()
```

Comment:
nit: it would be more idiomatic to use an atomic int instead of a mutex here. you locking a a primitive data type, not some arbitrary code block.

Reply:
Oh good point, forgot those existed